lib/logstash/inputs/snmptrap.rb

```diff
@@ -47,26 +47,38 @@ def register
     end
   end # def register
 
-  public
   def run(output_queue)
     begin
       # snmp trap server
       snmptrap_listener(output_queue)
     rescue => e
-      @logger.warn("SNMP Trap listener died", :exception => e, :backtrace => e.backtrace)
-      sleep(5)
-      retry
+      if !stop?
+        @logger.warn("SNMP Trap listener died", :exception => e, :backtrace => e.backtrace)
+        sleep(5)
```

Comment:
This would be required to be tracked too, imagine what would happen if you aim to stop when are into this sleep.